python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -644,49 +647,58 @@ class RandomForestClassifier(Base):
-                              coalescing-friendly
-               'BATCH_TREE_REORG' - similar to TREE_REORG but predicting
-                                    multiple rows per thread block
-        threshold : float
-                    threshold is used to for classification
-                    This is optional and required only while performing the
-                    predict operation on the GPU.
-                    It is applied if output_class == True, else it is ignored
-        num_classes : integer
+        predict_model : String (default = 'GPU')
+            'GPU' to predict using the GPU, 'CPU' otherwise. The GPU can only
+            be used if the model was trained on float32 data and `X` is float32
+            or convert_dtype is set to True.
+        output_class: boolean (default = True)
+            This is optional and required only while performing the
+            predict operation on the GPU.
+            If true, return a 1 or 0 depending on whether the raw
+            prediction exceeds the threshold. If False, just return
+            the raw prediction.
+        algo : string (default = 'BATCH_TREE_REORG')
+            This is optional and required only while performing the
+            predict operation on the GPU.
+            'NAIVE' - simple inference using shared memory
+            'TREE_REORG' - similar to naive but trees rearranged to be more
+                           coalescing-friendly
+            'BATCH_TREE_REORG' - similar to TREE_REORG but predicting
+                                 multiple rows per thread block
+        threshold : float (default = 0.5)
+            Threshold used for classification. Optional and required only
+            while performing the predict operation on the GPU.
+            It is applied if output_class == True, else it is ignored
+        num_classes : int (default = 2)
                       number of different classes present in the dataset
+        convert_dtype : bool, optional (default = False)
+            When set to True, the predict method will, when necessary, convert
+            the input to the data type which was used to train the model. This
+            will increase memory used for the method.
 
         Returns
         ----------
         y : NumPy
            Dense vector (int) of shape (n_samples, 1)
         """
-        if self.dtype == np.float64:
-            raise TypeError("GPU predict model only accepts float32 dtype"
-                            " as input, convert the data to float32 or "
-                            "use the CPU predict with `predict_model='CPU'`.")
 
-        elif predict_model == "CPU" or predict_model is None:
-            preds = self._predict_model_on_cpu(X)
+        if predict_model == "CPU":
+            preds = self._predict_model_on_cpu(X, convert_dtype)
+
+        elif self.dtype == np.float64 and convert_dtype is False:
+            warnings.watn("Using CPU based predict. \
```

Comment:
`warnings.warn`

I actually initially typed `warnings.warm` so somehow this is typo-prone ;)

Reply:
lol :)